python/dgl/graphbolt/impl/uniform_negative_sampler.py

```diff
@@ -86,20 +86,9 @@ def _sample_with_etype(self, node_pairs, etype=None, use_seeds=False):
             # Construct labels for all node pairs.
             pos_num = node_pairs.shape[0]
             neg_num = seeds.shape[0] - pos_num
-            labels = torch.cat(
-                (
-                    torch.ones(
-                        pos_num,
-                        dtype=torch.bool,
-                        device=seeds.device,
-                    ),
-                    torch.zeros(
-                        neg_num,
-                        dtype=torch.bool,
-                        device=seeds.device,
-                    ),
-                ),
-            )
+            labels = torch.empty(pos_num + neg_num, device=seeds.device)
```

Comment:
Why is using the default dtype better?

Reply:
When using bool as dtype will cause error  as followed during training  https://github.com/dmlc/dgl/blob/master/examples/sampling/graphbolt/link_prediction.py#L328:
```
Traceback (most recent call last):
  File "/home/ubuntu/dgl/examples/sampling/graphbolt/link_prediction_seeds.py", line 440, in <module>
    main(args)
  File "/home/ubuntu/dgl/examples/sampling/graphbolt/link_prediction_seeds.py", line 422, in main
    train(args, model, graph, features, train_set)
  File "/home/ubuntu/dgl/examples/sampling/graphbolt/link_prediction_seeds.py", line 337, in train
    loss = F.binary_cross_entropy_with_logits(logits, labels)
  File "/home/ubuntu/miniconda3/envs/dgl-dev-gpu-118/lib/python3.10/site-packages/torch/nn/functional.py", line 3165, in binary_cross_entropy_with_logits
    return torch.binary_cross_entropy_with_logits(input, target, weight, pos_weight, reduction_enum)
RuntimeError: Subtraction, the `-` operator, with two bool tensors is not supported. Use the `^` or `logical_xor()` operator instead.
```
So, we should use other dtype instead. I searched previous code and found labels was using default dtype in https://github.com/dmlc/dgl/blob/master/python/dgl/graphbolt/minibatch.py#L482-L484

Reply:
Thank you for the pointers